pkg/env/certificateRefresher_test.go

```diff
@@ -161,32 +161,51 @@ func TestRefreshingCertificate(t *testing.T) {
 
 	cannotPull := errors.New("Cannot pull certificates from keyvault")
 
+	// signal inline function is used to mock time.Ticker to pull the certificate
+	//
+	// func(n) prepares ticker function to tick exactly n-times
+	// returned function gets instantiated at the moment of use as ticker
+	// this is important for the tests to stay isolated
+	//
+	// func(start chan) is a ticker function which uses chan to signal when to
+	// start ticking. Flagging start is important for simulating the actual
+	// timeout of the real timer. Without start it would go all in and break
+	// testing.
+	signal := func(n int) func(chan struct{}) chan time.Time {
+		return func(start chan struct{}) chan time.Time {
+			s := make(chan time.Time)
+			go func() {
+				<-start
+				for i := 0; i < n; i++ {
+					s <- time.Now()
```

Comment:
Maybe just send empty time instead of "now" because we do not rely on time: will be more clear that time is not important here.